### ISSUE TITLE
Add support for non numeric values for spacing

### ISSUE THREAD

**Issue by nathancharles** [linkedin/archetype#104]
Currently trying to use non numeric values for spacing seems to throw an error.
@include horizontal-spacing(auto, $method: margin);

It would be nice to add support for non numeric values for spacing (auto, inherit, etc)

**Comment by eoneill** [comment]
I've re-thought this a few times and I keep landing on a relaxed API that allows essentially any valid CSS value. Unitless numbers will be multiplied out against the spacing config whereas numbers with units will just pass through untouched.

This would allow e.g.:
```scss
@include horizontal-spacing(auto 5); // `auto` is left alone, `5` is multiplied
@include horizontal-spacing(3 5px); //  `3` is multiplied, `5px` is left alone
```

I'll have a PR with these changes shortly.

**Comment by eoneill** [comment]
This is resolved with #105 and is available in release [v1.0.0.alpha.5](https://rubygems.org/gems/archetype/versions/1.0.0.alpha.5).
